### PR TITLE
Adds a random martial arts modifier to deathmatch + Adds a blacklisted map framework to modifiers

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request

Title.

Uses pickweight, to ensure rare martial arts stay rare. 

Common:

* Carp
* CQC

Uncommon:
* EVIL boxing

LEGENDARY:
* Wrestling
* UNNERFED PLASMA FIST
* Psychotic brawling (bath salts, rare because its a meme)

Does not work on meat tower.
## Why It's Good For The Game

Variety is good! Also, theres only one way to get martial arts in DM and thats meat tower. There is ALSO very little way to get evil boxing, mushpunch, plasmafist, etc. This allows people to go absolutely nuts in death match, and I think thats a good thing.
## Changelog
:cl:
add: New deathmatch modifier: "Random martial arts"
/:cl:

### CODE REVIEW

**Comment by nikothedude** [comment]
krav maga is excluded because its kinda really common outside deathmatch and its really basic

**Comment by necromanceranne** [comment]
You might want to have EVIL boxing grant master rank athletics skill or something, because baseline EVIL boxing without skill boosts is pretty weak.

**Comment by Admiral-Gigaton** [comment]
awww, no way to select which! oh well, at least now i can practice CQC

**Comment by nikothedude** [comment]
> awww, no way to select which! oh well, at least now i can practice CQC

theres no framework for modifier modifiers

File: code/modules/deathmatch/deathmatch_modifier.dm [hunk 1]
```diff
@@ -554,3 +568,33 @@
...
+	)
+
+/datum/deathmatch_modifier/martial_artistry/apply(mob/living/carbon/player, datum/deathmatch_lobby/lobby)
+	. = ..()
+
+	var/datum/martial_art/picked_art_path = pick_weight(weighted_martial_arts)
+	var/datum/martial_art/instantiated_art = new picked_art_path()
+
+	if (istype(instantiated_art, /datum/martial_art/boxing))
+		player.mind.adjust_experience(/datum/skill/athletics, SKILL_EXP_LEGENDARY)
```
Comment (necromanceranne) [inline code/modules/deathmatch/deathmatch_modifier.dm:596]: Wouldn't this autogrant people the athletics achievement?

Comment (nikothedude) [reply code/modules/deathmatch/deathmatch_modifier.dm:596]: it didn't do so locally but also it was a local test

Comment (nikothedude) [reply code/modules/deathmatch/deathmatch_modifier.dm:596]: there is no achievement for this

**Comment by Loafin34** [comment]
give them gripper gloves

**Comment by nikothedude** [comment]
> give them gripper gloves

why

**Comment by Admiral-Gigaton** [comment]
how does unnerfed plasma fist work? I wasn't around to see it, and i've only seen it once, post nerf.

**Comment by nikothedude** [comment]
> how does unnerfed plasma fist work? I wasn't around to see it, and i've only seen it once, post nerf.

it just lets you use apotheosis

**Pull request merged**